src/main/java/org/chsrobotics/lib/math/UtilityMath.java

```diff
@@ -207,4 +210,237 @@ public static Tuple2<Double> quadraticZeros(double coeffA, double coeffB, double
 
         return Tuple2.of(pos, neg);
     }
+
+    /**
+     * Takes in a series of numbers and returns their arithmetic mean.
+     *
+     * @param inputs The values to average. If empty, this will return {@code 0}.
+     * @return The arithmetic mean of the values.
+     */
+    public static double arithmeticMean(double... inputs) {
+        if (inputs.length == 0) return 0;
+
+        double sum = 0;
+
+        for (double value : inputs) sum += value;
+
+        return sum / inputs.length;
+    }
+
+    /**
+     * Takes in a series of numbers and returns their geometric mean.
+     *
+     * @param inputs The values to average. If empty, this will return {@code 0}.
+     * @return The geometric mean of the values.
+     */
+    public static double geometricMean(double... inputs) {
+        if (inputs.length == 0) return 0;
+
+        double product = 1;
+
+        for (double value : inputs) product = product * value;
+
+        return root(product, inputs.length);
+    }
+
+    /**
+     * Returns whether two values are equal to within an epsilon (calculated by their difference and
+     * expected from floating-point arithmetic). Essentially just "close enough".
+     *
+     * @param valueA The first number to compare.
+     * @param valueB The second number to compare.
+     * @param absoluteEpsilon The maximum allowed difference between the two to return true.
+     * @return Whether two values are "close enough" to each other.
+     */
+    public static boolean epsilonEqualsAbsolute(
```

Comment:
can we remove this if readding apache math?

Reply:
tbh I'd rather use our own methods, especially with epsilonEqualsProportion (I'd like a quick counterpart for it instead of having to know that Precision exists)

Reply:
that’s fair, but what about things like sin and cos